client/libra-dev/src/event.rs

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    data::{LibraEvent, LibraEventType, LibraPaymentEvent, LibraStatus},
+    error::*,
+};
+use libra_types::{
+    account_address::ADDRESS_LENGTH, account_config::ReceivedPaymentEvent,
+    account_config::SentPaymentEvent, contract_event::ContractEvent, event::EventKey,
+    event::EVENT_KEY_LENGTH, language_storage::TypeTag,
+};
+use std::convert::TryFrom;
+use std::ffi::CString;
+use std::ops::Deref;
+use std::slice;
+
+#[no_mangle]
+pub unsafe extern "C" fn libra_LibraEvent_from(
+    buf_key: *const u8,
+    len_key: usize,
+    buf_data: *const u8,
+    len_data: usize,
+    buf_type_tag: *const u8,
+    len_type_tag: usize,
+    out: *mut *mut LibraEvent,
+) -> LibraStatus {
+    clear_error();
+    let buffer_key: &[u8] = slice::from_raw_parts(buf_key, len_key);
+    let buffer_data: &[u8] = slice::from_raw_parts(buf_data, len_data);
+    let buffer_type_tag: &[u8] = slice::from_raw_parts(buf_type_tag, len_type_tag);
```

Comment:
`from_raw_parts` requires the pointer to be non-NULL which is not checked here. Perhaps bail if NULL as you do for some of the other functions?